src/util/uint.rs

```diff
@@ -73,18 +73,31 @@ macro_rules! construct_uint {
             }
 
             /// Create an object from a given unsigned 64-bit integer
+            #[inline]
             pub fn from_u64(init: u64) -> Option<$name> {
                 let mut ret = [0; $n_words];
                 ret[0] = init;
                 Some($name(ret))
             }
 
             /// Create an object from a given signed 64-bit integer
+            #[inline]
             pub fn from_i64(init: i64) -> Option<$name> {
                 assert!(init >= 0);
                 $name::from_u64(init as u64)
             }
 
+            /// Creates big integer value from a byte slice array using
+            /// big-endian encoding
+            pub fn from_be_bytes(bytes: [u8; $n_words * 8]) -> $name {
+                use super::endian::slice_to_u64_be;
+                let mut slice = [0u64; $n_words];
+                for n in 0..$n_words {
+                    slice[$n_words - n - 1] = slice_to_u64_be(&bytes[(n * 8)..(n * 8 + 8)]);
+                }
```

Comment:
We could avoid all that hard to read byte arithmetic by using iterators.

```suggestion
                slice.iter_mut()
                    .rev()
                    .zip(bytes.chunks(8))
                    .for_each(|(word, bytes)| *word = slice_to_u64_be(bytes));
```

Reply:
(FWIW they result in the same ASM so perf wise they're the same: https://godbolt.org/z/9rnG9f)

Reply:
Interesting, indexing is usually slower.